src/shogun/lib/SGVector.h

```diff
@@ -474,6 +477,25 @@ template<class T> class SGVector : public SGReferencedData
 		 * @return matrix
 		 */
 		static void convert_to_matrix(T*& matrix, index_t nrows, index_t ncols, const T* vector, int32_t vlen, bool fortran_order);
+
+		template<class Archive>
+		void cereal_save(Archive & ar) const
+		{
+			ar(vlen);
+			for (index_t i = 0; i < vlen; ++i)
+				ar(vector[i]);
+		}
+
+		template<class Archive>
+		void cereal_load(Archive & ar)
+		{
+			unref();
```

Comment:
With discussions in #3375.
I think we need the unref() - because we allocated new space for loaded data and the old ones are gone.
For the same reason I think we need a new RefCount.
However, I cannot do it now without appropriate reset method or serialization method in SGReferenceData.

The old load method was like:
```
	SGVector<T> vec;
	loader->get_vector(vec.vector, vec.vlen);
	copy_data(vec);
	copy_refcount(vec);
	ref();
```